src/store/middlewares/login.js

```diff
@@ -17,30 +14,29 @@ const loginMiddleware = store => next => (action) => { // eslint-disable-line ma
           data: { isHarwareWalletConnected: false },
         });
 
-        ipc.on('ledgerConnected', () => {
+        ipc.on('hwConnected', (event, { model }) => {
```

Comment:
IMO it doesn't make sense why listening to hwWallet events is in `login` as it has nothing to do with login. It should be rather in `src/store/middlewares/hwWallet.js`

Reply:
I think it should be part of the login process but it doesn't need to be part of middleware and it should happen inside the login action.

Reply:
I think it should be part of the login process but it doesn't need to be part of middleware and it should happen inside the login action.

Reply:
Listening to `hwConnected` event **cannot** be inside `login` action. We first need to detect/display that the device was connected, and only then the user can log in.